src/main/java/org/openrewrite/java/migrate/maven/UseMavenCompilerPluginReleaseConfiguration.java

```diff
@@ -114,7 +115,8 @@ private boolean currentNewerThanProposed(@SuppressWarnings("OptionalUsedAsFieldO
         }
     }
 
-    private boolean hasJavaVersionProperty(Xml.Document xml) {
-        return !FindProperties.find(xml, "java.version").isEmpty();
+    private boolean hasJavaVersionProperty(Xml.Document xml, ExecutionContext ctx) {
+        // If the instance changes, is because it added some SearchResult
+        return new FindProperties("java\\.version").getVisitor().visitNonNull(xml, ctx) != xml;
```

Comment:
Thank you very much! This was the alternative I was thinking of, but I thought that this would resolve the parent pom in run time, being less performant, so I decided to go with just tuning the existing approach.
Much appreciated! 😄